client/src/components/Game/Timer.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import { withFirebase } from '../Firebase';
+import { withRouter } from 'react-router';
+import { compose } from 'recompose';
+
+class Timer extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+
+    }
+  }
+
+  async componentDidMount() {
+    const { id } = this.props.match.params;
+    const game = this.props.firebase.game(id);
+
+    game.get()
+    .then(docSnapshot => {
+      if(docSnapshot.exists) {
+        this.unsubscribe = game.onSnapshot((snapshot) => {
+          if(snapshot.data().gameStartTime !== null && snapshot.data().currentPlayerIndex === 0) {
+            this.setState({
+              currentPlayerIndex: snapshot.data().currentPlayerIndex + 1,
+              startTime: snapshot.data().gameStartTime.seconds,
+              timePerTurnInSeconds: parseInt(snapshot.data().timeLimit, 10) * 60,
+            })
+            this.setState({
+              timeTurnWillEnd: (this.state.timePerTurnInSeconds + this.state.startTime),
+            })
+            this.setState({
+              days: Math.floor((((this.state.timeTurnWillEnd - Math.floor(new Date().getTime() / 1000)) / 60) / 60)/ 24),
```

Comment:
Looks like a lot of repetitive code. Can we increase code reuse by putting these calculations into helper function(s)?